packages/fixed-decimal/src/FixedDecimal.ts

```diff
@@ -0,0 +1,470 @@
+import type BN from 'bn.js'
+
+/**
+ * Parses a string into the constructor args for a {@link FixedDecimal}.
+ *
+ * Doesn't do any validation of the string - if it's malformed the `BigInt`
+ * construction will fail.
+ * @param value The value represented as a fixed decimal string.
+ * @param decimalPlaces The number of decimal places the result should have.
+ * @returns
+ */
+const parseFixedDecimalString = (
+  value: string,
+  decimalPlaces?: number
+): FixedDecimalCtorArgs => {
+  let [whole, decimal] = value.split('.')
+  decimal = decimal ?? ''
+  if (decimalPlaces !== undefined) {
+    decimal = decimal.padEnd(decimalPlaces, '0').substring(0, decimalPlaces)
+  }
+  return {
+    value: BigInt(`${whole}${decimal}`),
+    decimalPlaces: decimalPlaces ?? decimal.length
+  }
+}
+
+type FixedDecimalCtorArgs = {
+  value: bigint
+  decimalPlaces: number
+}
+
+/**
+ * A custom options type for the custom toLocaleString() implementation of
+ * {@link FixedDecimal} that only allows the supported subset of the
+ * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat Intl.NumberFormat options}.
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat MDN documentation for Intl.NumberFormat}
+ */
+type FixedDecimalFormatOptions = {
+  /**
+   * Whether to use grouping separators, such as thousands separators or thousand/lakh/crore separators.
+   *
+   * Note: Does not support `'always'`, `'auto'`, or `'min2'`
+   * @defaultValue `true`
+   */
+  useGrouping?: boolean
+  /**
+   * The minimum number of fraction digits to use.
+   * @defaultValue `0`
+   */
+  minimumFractionDigits?: number
+  /**
+   * The maximum number of fraction digits to use.
+   * @defaultValue `this.decimalPlaces` (include all decimal places)
+   */
+  maximumFractionDigits?: number
+  /**
+   * How decimals should be rounded.
+   *
+   * Possible values are:
+   *
+   * `'ceil'`
+   *    > Round toward +∞. Positive values round up.
+   *      Negative values round "more positive".
+   *
+   * `'floor'`
+   *    > Round toward -∞. Positive values round down.
+   *      Negative values round "more negative".
+   *
+   * `'trunc'` (default)
+   *    > Round toward 0. This _magnitude_ of the value is always
+   *      reduced by rounding. Positive values round down.
+   *      Negative values round "less negative".
+   *
+   * `'halfExpand'`
```

Comment:
I believe this is called "Half Up" usually

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#halfexpand

halfUp is confusing because "up" for negatives is to a "less negative" number

Reply:
Didn't realize they rebranded it https://docs.oracle.com/javase/8/docs/api/java/math/RoundingMode.html. I guess "commercial rounding" is another term https://en.wikipedia.org/wiki/Rounding

Reply:
> [citation needed]

😜